starboard/tools/testing/test_runner.py

```diff
@@ -163,12 +164,24 @@ class TestLauncher(object):
 
   def __init__(self, launcher):
     self.launcher = launcher
-    self.runner_thread = threading.Thread(target=self._Run)
 
     self.return_code_lock = threading.Lock()
     self.return_code = 1
 
   def Start(self):
+    if hasattr(self.launcher, "Run2"):
+      if hasattr(self.launcher, "InitDevice"):
+        self.launcher.InitDevice()
+      if abstract_launcher.ARG_NOINSTALL not in self.launcher.launcher_args \
+        and hasattr(self.launcher, "Deploy"):
+        self.launcher.Deploy()
+        assert hasattr(self.launcher, "CheckPackageIsDeployed")
```

Comment:
I would recommend moving the assert up by one line, before making the `Deploy` call.

Reply:
Done.